topdown/crypto.go

```diff
@@ -126,6 +131,45 @@ func builtinCryptoX509ParseCertificateRequest(a ast.Value) (ast.Value, error) {
 	return ast.InterfaceToValue(x)
 }
 
+func builtinCryptoX509ParseRSAPrivateKey(_ BuiltinContext, args []*ast.Term, iter func(*ast.Term) error) error {
+
+	a := args[0].Value
+	input, err := builtins.StringOperand(a, 1)
+	if err != nil {
+		return err
+	}
+
+	invalid := ast.BooleanTerm(false)
+
+	// get the raw private key
+	rawKey, err := getRSAPrivateKeyFromString(string(input))
+	if err != nil {
+		return iter(invalid)
+	}
+
+	rsaPrivateKey, err := jwk.New(rawKey)
+	if err != nil {
+		return iter(invalid)
+	}
+
+	jsonKey, err := json.Marshal(rsaPrivateKey)
+	if err != nil {
+		return iter(invalid)
+	}
+
+	var x interface{}
+	if err := util.UnmarshalJSON(jsonKey, &x); err != nil {
+		return err
+	}
```

Comment:
Why do we need to go through JSON here? I suppose it's a shortcut, we could also declare the object as-is, couldn't we? But I guess this matches what we do when the key is expected in object form (i.e. marshal it to *PrivateKey), so this is probably OK 😄

Reply:
Yes it's more like a shortcut, and the following `ast.InterfaceToValue()` needs it in that form and I didn't step into very much for the `InterfaceToValue`  😄